titus-server-master/src/main/java/com/netflix/titus/master/jobmanager/endpoint/v3/grpc/DefaultJobManagementServiceGrpc.java

```diff
@@ -171,6 +174,8 @@ public DefaultJobManagementServiceGrpc(GrpcMasterEndpointConfiguration configura
         this.authorizationService = authorizationService;
         this.titusRuntime = titusRuntime;
         this.schedulingService = schedulingService;
+        this.observeJobsScheduler = Schedulers.from(ExecutorsExt.instrumentedFixedSizeThreadPool(
```

Comment:
Personal preference, but I would recommend against using a fixed size executor here\*.   Depending on how spiky your traffic is, a better way to bound the traffic is to keep track of the number of active RPCs, and fail new RPCs at your application layer.  (by keeping track of latency, deadline expirations, etc.)    Feel free to keep it as is, but just wanted to make aware that fixed thread pools have some risky side effects that are hard to debug.

\* I know I said otherwise on the S/O answer, but there is more nuance needed for an accurate answer (but which doesn't fit well into the S/O format).

Reply:
Makes sense. There's a lot of nuance here, and even with a concurrency limiter at the application code level (RPC handlers) we would still want to limit the size of this thread pool for isolation purposes, and to protect other components running in the same JVM. This was the intent here, we will look into adding more protection (concurrency limits) for RPCs in the future.